tests/test_slackrequest.py

```diff
@@ -2,42 +2,71 @@
 import json
 import os
 
+def test_http_headers(mocker):
+    requests = mocker.patch('slackclient._slackrequest.requests')
+    request = SlackRequest()
+
+    request.do('xoxb-123', 'chat.postMessage', {'text': 'test', 'channel': '#general'})
+    args, kwargs = requests.post.call_args
+
+    assert None != kwargs['headers']['user-agent']
+
+def test_custom_user_agent(mocker):
+    requests = mocker.patch('slackclient._slackrequest.requests')
+    request = SlackRequest()
+
+    request.append_user_agent("fooagent1","0.1")
+    request.append_user_agent("baragent/2","0.2")
+
+    request.do('xoxb-123', 'chat.postMessage', {'text': 'test', 'channel': '#general'})
+    args, kwargs = requests.post.call_args
+
+    # Verify user-agent includes both default and custom agent info
+    assert "slackclient" in kwargs['headers']['user-agent']
+    assert "fooagent1" in kwargs['headers']['user-agent']
+
+    # verify escaping of slashes in custom agent name
+    assert "baragent:2" in kwargs['headers']['user-agent']
```

Comment:
You're note verifying that the right value is associated with the key, only that the key is present (and appropriately mutilated)